test/aws.storage.test.js

```diff
@@ -0,0 +1,140 @@
+const assert = require('assert');
+const sinon = require('sinon');
+const proxyquire = require('proxyquire');
+
+const conf = require('../server/config.js');
+conf.notLocalHost = true;
+
+let redisStub = {};
+let exists = sinon.stub();
+let hget = sinon.stub();
+let hmset = sinon.stub();
+let expire = sinon.stub();
+let del = sinon.stub();
+
+redisStub.createClient = function() {
+  return {
+    on: sinon.spy(),
+    exists: exists,
+    hget: hget,
+    hmset: hmset,
+    expire: expire,
+    del: del
+  }
+}
+
+let fsStub = {};
+fsStub.statSync = sinon.stub();
+fsStub.createReadStream = sinon.stub();
+fsStub.createWriteStream = sinon.stub();
+fsStub.unlinkSync = sinon.stub();
+
+let logStub = {};
+logStub.info = sinon.stub();
+logStub.error = sinon.stub();
+
+let s3Stub = {};
+s3Stub.headObject = sinon.stub();
+s3Stub.getObject = sinon.stub();
+s3Stub.upload = sinon.stub();
+s3Stub.deleteObject = sinon.stub();
+
+let awsStub = {
+    S3: function() {
+        return s3Stub;
+    }
+}
+
+const storage = proxyquire('../server/storage', {
+  'redis': redisStub,
+  'fs': fsStub,
+  './log.js': function() { return logStub },
+  'aws-sdk': awsStub
+});
+
+describe('Testing Length using aws', function() {
+  it('Filesize returns properly if id exists', function() {
+    s3Stub.headObject.callsArgWith(1, null, {ContentLength: 1});
+    return storage.length('123')
+                  .then(reply => assert(reply === 1))
+                  .catch(err => assert.fail())
+  })
+
+  it('Filesize fails if the id does not exist', function() {
+    s3Stub.headObject.callsArgWith(1, new Error(), null);
+    return storage.length('123')
+                  .then(reply => assert.fail())
+                  .catch(err => assert(1))
+  })
+});
+
+describe('Testing Get using aws', function() {
+  it('Should not error out when the file exists', function() {
+    s3Stub.getObject.returns({
+      createReadStream: function() { return 1; }
+    });
+    assert(storage.get('123') === 1);
+  })
+
+  it('Should error when the file does not exist', function() {
+    s3Stub.getObject.returns({
+      createReadStream: function() { return null; }
+    });
+    assert(storage.get('123') === null);
+  })
+});
+
+describe('Testing Set using aws', function() {
+  it('Should pass when the file is successfully uploaded and no bitly key', function() {
+    conf.bitly_key = null;
+    s3Stub.upload.callsArgWith(1, null, {});
+    return storage.set('123', {}, 'Filename.moz', 'url.com')
+           .then(reply => assert(reply.url === 'url.com' && reply.uuid !== null))
```

Comment:
Since this function has multiple side effects it's useful to assert that the ones we expect to get called actually were. For example, setting the expire time in redis. We can use sinon to assert that it was called, and with the arguments we expect:
```js
assert(expire.calledOnce)
assert(expire.calledWith('123', 86400000))
```
ref: http://sinonjs.org/releases/v2.3.4/spies/

For this to work you might need to create a [beforeEach](http://mochajs.org/#hooks) to call `expire.reset()` etc.